web/src/muted_users_ui.ts

```diff
@@ -13,6 +13,10 @@ export function rerender_for_muted_user(): void {
         msg_list.update_muting_and_rerender();
     }
 
+    for (const msg_list_data of message_lists.non_rendered_data()) {
+        msg_list_data.update_items_for_muting();
+    }
```

Comment:
This one may or may not be necessary; it's possible we want to defer doing this step with a comment here, and jsut make sure `update_items_for_muting` gets called when waking up a cached `MessageListData` object.

Well, this one is probably fine because muting users is really rare, but I'd worry a similar thing for muting topics would be slow.

Reply:
Makes sense!

Updated muted items here:
```js
        let msg_data = message_list_data_cache.get(filter);
        let msg_data_is_prepopulated = true;
        if (msg_data === undefined) {
            // If we don't have a cached message list for the narrow, we
            // need to construct one from scratch.
            msg_data = new MessageListData({
                filter,
                excludes_muted_topics,
            });
            msg_data_is_prepopulated = false;
        } else {
            // Do updates to msg_data that might change the message we select here.
            msg_data.update_items_for_muting();
        }
```

Reply:
Okay, modified to return a copy.